### PR TITLE
Fix mirror token

### PR DESCRIPTION
**Pull request by zjb0807**
Closes: #1815

### CODE REVIEW

**Review by xlc — COMMENTED** [review]
One issue is that if we want upgrade predeployed token contract, we will need to upgrade to all of them instead of just one.

Another solution is to modify bodhi.js side to special case handling the code access code for predeployed tokens.

File: modules/evm/src/lib.rs [hunk 1]
```diff
@@ -319,6 +319,20 @@ pub mod module {
 		pub ref_count: u32,
 	}
 
+	#[derive(Clone, Copy, Eq, PartialEq, RuntimeDebug, Encode, Decode, MaxEncodedLen, TypeInfo)]
+	#[cfg_attr(feature = "std", derive(Serialize, Deserialize))]
+	pub enum AccountRole {
+		Account,
+		Contract,
+		Developer,
```
[the remark this answers is not in the document]
Comment (xlc) [reply modules/evm/src/lib.rs:327]: also the contract role is only applied for genesis contracts, which is a bit confusing.

**Comment by zjb0807** [comment]
> Another solution is to modify bodhi.js side to special case handling the code access code for predeployed tokens.

Avoid replacing addresses by rules(`handle_mirrored_token`), which will reduce some unknown errors and risks. The contract does not exist but has been replaced with the predeployed token contract

**Review by xlc — APPROVED** [review]
This means we need to manually manage predeploy contract version, which is fine, just need to remember to do it with new supported tokens, LP, etc.

**Review by ermalkaleci — APPROVED** [review]

**Pull request merged**